### PR TITLE
Fix StdScaler overflow for large-valued inputs

### PR DESCRIPTION
**Pull request by K2alyan**
*Issue #, if available:* #3070

## What was happening

`StdScaler` computes weighted variance by squaring the centered values directly in float32. For inputs with values around 1e20 (which isn't unusual in financial or physical datasets), `(x - mean)^2` lands around 1e40 — past float32's ceiling of ~3.4e38. The result is `inf` scale, which cascades into `NaN` model weights and predictions.

The minimal reproducer from the issue:

```python
import torch
from gluonts.torch.scaler import StdScaler

x = torch.empty(5, 10).normal_(mean=0, std=1e20)
_, _, scale = StdScaler(keepdim=True)(x, torch.ones_like(x))
print(scale)  # tensor([[inf], [inf], [inf], [inf], [inf]])
```

After this fix:

```python
print(scale)  # tensor([[9.8e+19], [1.0e+20], ...])  — finite, correct
```

## The fix

Cast to float64 before squaring, take the sqrt there, then cast back. The standard deviation itself (~1e20) is representable in float32; it's the intermediate variance (~1e40) that isn't. The cast is inside `torch.no_grad()` so there's no gradient overhead.

```python
# before
variance = (((data - loc) * weights) ** 2).sum(...) / denominator

# after
variance = (
    (data.double() - loc.double()).pow(2) * weights
).sum(...) / denominator
scale = torch.sqrt(variance + self.minimum_scale).to(data.dtype)
```

The rewrite also corrects the formula: the original code squared `weights` alongside the deviations (`((data - loc) * weights) ** 2`). For the binary 0/1 indicators GluonTS uses this produces the same numbers, but it doesn't match the weighted-variance definition. Changed to `(data - loc).pow(2) * weights`.

## Tests

Added three `StdScaler` tests that were missing from the file:

- constant series (variance=0, scale falls back to sqrt(minimum_scale), output is zero)
- partial observation (loc and scale computed from observed window only)
- the exact overflow reproducer from #3070 — asserts `scale` and `scaled` are finite

**Please tag this pr with at least one of these labels to make our release process faster:** bug fix

### CODE REVIEW

**Review by lostella — APPROVED** [review]
Thanks @K2alyan!
